src/spatial/utils.jl

```diff
@@ -0,0 +1,217 @@
+using LightGraphs
+"""
+A file with types, structs and functions for spatial simulations
+"""
+
+"""
+stores info for a spatial jump
+"""
+struct SpatialJump{J}
+    src::J    # source location
+    jidx::Int # index of jump as a hop or reaction
+    dst::J    # destination location
+end
+
+############ abstract spatial system struct ##################
+"""
+Contains all info about the topology of the system
+"""
+abstract type AbstractSpatialSystem end
+
+# Interface:
+# num_sites(spatial_system) = total number of sites
+# neighbors(spatial_system, site) = an iterator over the neighbors of site
+# num_neighbors(spatial_system, site) = number of neighbors of site
+
+################### LightGraph ########################
+num_sites(graph::AbstractGraph) = LightGraphs.nv(graph)
+neighbors(graph::AbstractGraph, site) = LightGraphs.neighbors(graph, site)
```

Comment:
If you are `using LightGraphs` why do you need to define this? Calling `neighbors(graph,site)` should call `LightGraphs.neighbors(graph,site)` already.

Reply:
For some reason I get an error if I don't do it. Not sure why.

Reply:
Are you trying to set
```julia
 neighbors(graph::AbstractGraph, site) = neighbors(graph, site) 
```
You don't need to do that...

If not, where / what error are you getting if you just remove this definition?